components/org.wso2.carbon.identity.conditional.auth.functions.http/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/http/CookieFunctionImpl.java

```diff
@@ -46,8 +47,18 @@
     private static final Log log = LogFactory.getLog(CookieFunctionImpl.class);
 
     @Override
-    public void setCookie(JsServletResponse response, String name, String value, Map<String, Object> properties) {
+    public void setCookie(JsServletResponse response, String name, Object... params) {
 
+        Map<String, Object> properties = new HashMap<>();
```

Comment:
We create HashMap unnecessarily. and reassign to different one at line 58. Instead we can init with null and then check for null at line 62.